src/main/java/core/basesyntax/strategy/OperationStrategyImpl.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.operation.OperationHandler;
+import java.util.Map;
+
+public class OperationStrategyImpl implements OperationStrategy {
+    private final Map<FruitTransaction.Operation, OperationHandler> operationHandlersMap;
+
+    public OperationStrategyImpl(Map<FruitTransaction.Operation,
+            OperationHandler> operationHandlersMap) {
+        this.operationHandlersMap = operationHandlersMap;
+    }
+
+    @Override
+    public OperationHandler get(FruitTransaction.Operation type) {
+        return operationHandlersMap.get(type);
```

Comment:
Consider adding a check for null return values to handle cases where the requested operation type is not found in the map. This can prevent potential `NullPointerException` and provide a more informative error message.